src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -307,6 +325,70 @@ private BoundExpression CreateMethodGroupConversion(CSharpSyntaxNode syntax, Bou
             return new BoundConversion(syntax, group, conversion, @checked: false, explicitCastInCode: isCast, constantValueOpt: ConstantValue.NotAvailable, type: destination, hasErrors: hasErrors) { WasCompilerGenerated = source.WasCompilerGenerated };
         }
 
+        private BoundExpression CreateTupleConversion(CSharpSyntaxNode syntax, BoundExpression source, Conversion conversion, TypeSymbol destination, DiagnosticBag diagnostics)
+        {
+            // We have a successful tuple conversion; rather than producing a node 
+            // which is a conversion on top of a tuple literal, tuple conversion is an element-wise conversion of arguments.
+
+            var sourceTuple = (BoundTupleLiteral)source;
+            var targetType = (NamedTypeSymbol)destination;
+
+            Debug.Assert(conversion.Kind ==  ConversionKind.ImplicitTuple || conversion.Kind == ConversionKind.ImplicitNullable);
+            Debug.Assert((conversion.Kind == ConversionKind.ImplicitNullable) == targetType.IsNullableType());
+
+            bool hasErrors = false;
+            NamedTypeSymbol destinationWithoutNullable = null;
+
+            if (conversion.Kind == ConversionKind.ImplicitNullable)
+            {
+                destinationWithoutNullable = (NamedTypeSymbol)targetType.GetNullableUnderlyingType();
+                targetType = destinationWithoutNullable;
+            }
+
+            TupleTypeSymbol destTupleType;
+            if (targetType.IsTupleType)
+            {
+                destTupleType = (TupleTypeSymbol)targetType;
+                targetType = destTupleType.UnderlyingTupleType;
+            }
+
+            var arguments = sourceTuple.Arguments;
+            Debug.Assert(Compilation.IsTupleUnderlyingType(targetType, arguments.Length), "converting a tuple literal to incompatible type?");
+
+            var convertedArguments = ArrayBuilder<BoundExpression>.GetInstance(arguments.Length);
+            var targetElementTypes = ArrayBuilder<TypeSymbol>.GetInstance(arguments.Length);
+
+            TupleTypeSymbol.GetElementTypes(targetType, targetElementTypes);
```

Comment:
I would expect a `Get` method to return a value.  This instead is populating a builder.  Consider renaming to `AppendElementTypes` or `PopulateElementTypes`.